ai-ml/jark-stack/terraform/karpenter.tf

```diff
@@ -0,0 +1,22 @@
+resource "aws_iam_policy" "karpenter_controlloer_policy" {
+  description = "Additional IAM policy for Karpenter controller"
+  policy      = data.aws_iam_policy_document.karpenter_controller_policy.json
+}
+
+data "aws_iam_policy_document" "karpenter_controller_policy" {
+  statement {
+    actions = [
+      "ec2:RunInstances",
+      "ec2:CreateLaunchTemplate",
+    ]
+    resources = ["*"]
+    effect    = "Allow"
+    sid       = "Karpenter"
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_controller_policy_attachment" {
+  # name       = "karpenter-controller-policy-attachment"
+  role      = module.eks_blueprints_addons.karpenter.iam_role_name
+  policy_arn = aws_iam_policy.karpenter_controlloer_policy.arn
+}
```

Comment:
Move this to `addons.tf` and provide the policy_document as value for the `karpenter.source_policy_document`. The add-on will merge this policy into the default policy. See https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/c003cad83f930a16395a5e4cddd7daa1d74c76a3/main.tf#L2802 for reference.